src/Avalonia.Controls/Slider.cs

```diff
@@ -81,57 +121,83 @@ public double TickFrequency
+            _pointerMovedDispose?.Dispose();
 
             _decreaseButton = e.NameScope.Find<Button>("PART_DecreaseButton");
             _track = e.NameScope.Find<Track>("PART_Track");
             _increaseButton = e.NameScope.Find<Button>("PART_IncreaseButton");
 
+            if (_track != null)
+            {
+                _track.IsThumbDragHandled = true;
+            }
+
             if (_decreaseButton != null)
             {
-                _decreaseButton.Click += DecreaseClick;
+                _decreaseButtonPressDispose = _decreaseButton.AddDisposableHandler(PointerPressedEvent, TrackPressed, RoutingStrategies.Tunnel);
+                _decreaseButtonReleaseDispose = _decreaseButton.AddDisposableHandler(PointerReleasedEvent, TrackReleased, RoutingStrategies.Tunnel);
             }
 
             if (_increaseButton != null)
             {
-                _increaseButton.Click += IncreaseClick;
+                _increaseButtonSubscription = _increaseButton.AddDisposableHandler(PointerPressedEvent, TrackPressed, RoutingStrategies.Tunnel);
+                _increaseButtonReleaseDispose = _increaseButton.AddDisposableHandler(PointerReleasedEvent, TrackReleased, RoutingStrategies.Tunnel);
             }
+
+            _pointerMovedDispose = this.AddDisposableHandler(PointerMovedEvent, TrackMoved, RoutingStrategies.Tunnel);
         }
 
-        private void DecreaseClick(object sender, RoutedEventArgs e)
+        private void TrackMoved(object sender, PointerEventArgs e)
         {
-            ChangeValueBy(-LargeChange);
+            if (_isDragging)
+            {
+                MoveToPoint(e.GetCurrentPoint(_track));
+            }
         }
 
-        private void IncreaseClick(object sender, RoutedEventArgs e)
+        private void TrackReleased(object sender, PointerReleasedEventArgs e)
         {
-            ChangeValueBy(LargeChange);
+            _isDragging = false;
         }
 
-        private void ChangeValueBy(double by)
+        private void TrackPressed(object sender, PointerPressedEventArgs e)
         {
-            if (IsSnapToTickEnabled)
+            if (e.GetCurrentPoint(this).Properties.IsLeftButtonPressed)
             {
-                by = by < 0 ? Math.Min(-TickFrequency, by) : Math.Max(TickFrequency, by);
+                MoveToPoint(e.GetCurrentPoint(_track));
+                _isDragging = true;
             }
+        }
 
-            var value = Value;
-            var next = SnapToTick(Math.Max(Math.Min(value + by, Maximum), Minimum));
-            if (next != value)
-            {
-                Value = next;
-            }
+        private void MoveToPoint(PointerPoint x)
+        {
+            var orient = Orientation == Orientation.Horizontal;
+            var pointDen = orient ? _track.Bounds.Width : _track.Bounds.Height;
+            var pointNum = orient ? x.Position.X : x.Position.Y;
+            var logicalPos = MathUtilities.Clamp(pointNum / pointDen, 0.0d, 1.0d);
```

Comment:
This one has a possible division by zero (when for whatever reason track has 0 size).

Reply:
i perhaps could pass `double.Epsilon` on pointDen instead for this edge case?

Reply:
That can work, just so we can avoid weird behavior when people try to restyle.